app/views/pages/edit_item.js.erb

```diff
@@ -1,42 +1,69 @@
 $('body').css('overflow', 'hidden');
 $("<%= escape_javascript(render 'pages/edit_item', item: item) %>")
   .appendTo(".container");
-
 $('.overlay').show();
 
-/* select style */
-var select = $('.overlay select#page_id');
-function setSelectStyle() {
-  var selected = select.children().filter(':selected');
-  if (selected.val()) {
-    select.css('color', '#2E8B57');
-    select.css('font-style', '');
-  } else {
-    select.css('color', '#606060');
-    select.css('font-style', 'italic');
-  }
-}
+{
+  document.querySelector('.overlay #item_name').focus();
 
-setSelectStyle();
+  var editItemName = document.querySelector('.edit-item input[name="item[name]"]');
+  var itemName = document.querySelector('.edit-item .items-list-item-name span');
+  var itemNameKeyUp = editItemName.addEventListener('keyup', function (ev) {
+    itemName.innerText = ev.target.value;
+  });
 
-select.on('change', function () {
-  setSelectStyle();
-});
+  var editItemCategory = document.querySelector('.edit-item select[name="item[category_id]"]');
+  var itemCategory = document.querySelector('.edit-item .items-list-item');
+  var itemCategoryChange = editItemCategory.addEventListener('change', function (ev) {
+    var classNames = {};
+    <% Category.all.each do |ctgy| %>
+      classNames[<%= ctgy.id %>] = "<%= CategoriesHelper::css_class_name(ctgy) %>";
+    <% end %>
+    itemCategory.className = 'items-list-item ' + classNames[ev.target.value];
+  });
 
-$('#item_page_link_to').focus();
+  var editLinkToPage = document.querySelector('.edit-item select[name="link_to_page"]');
 
-function closeDialog(event) {
-  event.preventDefault();
-  $('.overlay').remove();
-  $('body').css('overflow', 'initial');
-  $(window).off('keyup', closeDialogOnEscape);
-}
+  if (editLinkToPage) {
+    var itemElement = document.querySelector('.edit-item .items-list-item');
+    var itemLinkToPage = getItemLinkToPage(itemElement);
 
-function closeDialogOnEscape (ev) {
-  if (ev.which === 27) {
-    closeDialog(ev);
+    var itemLinkToPageChange = editLinkToPage.addEventListener('change', function (ev) {
```

Comment:
Missing event listener removal in `closeDialog()` function if `editLinkToPage` exists
```javascript
$(editLinkToPage).off('change', itemLinkToPageChange);
```